packages/ingest/test/test-FtpProviderClientFailure.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const test = require('ava');
+const JSFtp = require('jsftp');
+
+const FtpProviderClient = require('../FtpProviderClient');
+
+const executeRawFtpCommand = async (cmd) => {
+  const jsftp = new JSFtp({
+    host: '127.0.0.1',
+    port: 21,
+    user: 'testuser',
+    pass: 'testpass'
+  });
+  return new Promise((resolve, reject) => {
+    jsftp.raw(cmd, (err, data) => {
+      if (err) reject(err);
+      resolve(data);
+    });
+  });
+};
+
+test.before(async () => {
+  // await when fixed
+  executeRawFtpCommand('chmod 400 -R forbidden/file.txt');
+});
+
+test.after.always(async () => {
+  // await when fixed
+  executeRawFtpCommand('chmod 644 -R forbidden/file.txt');
+});
+
+test.skip('FtpProviderClient throws an error when listing a non-permitted directory', async (t) => {
+  // TODO: update cumuluss/vsftpd to allow `chmod` to work in the setup for this test.
+  const myFtpProviderClient = new FtpProviderClient({
+    host: '127.0.0.1',
+    username: 'testuser',
+    password: 'testpass',
+    useList: true
+  });
+
+  await t.throwsAsync(myFtpProviderClient.list('forbidden/file.txt'),
+    'FTP Code 451: Could not retrieve a file listing for forbidden/file.txt.'
```

Comment:
Do we want to test against the full error message? You could test against a regex that ensures the error code and file name show up in the error message 🐜

Reply:
`throwsAsync` takes an error message to compare with, though. I guess I could catch it and test it manually.

Reply:
Yeah, looks like the AVA docs discuss that

```
const promise = Promise.reject(new TypeError('🦄'));

test('rejects', async t => {
	const error = await t.throwsAsync(promise);
	t.is(error.message, '🦄');
});
```

Definitely not necessary

Reply:
7e9d0ed26